lib/parse/dep-graph.ts

```diff
@@ -3,15 +3,19 @@ import type { MavenGraph, MavenGraphNode } from './types';
 import { DepGraph, DepGraphBuilder, PkgInfo } from '@snyk/dep-graph';
 import { parseDependency } from './dependency';
 
+const TEST_SCOPE = 'test';
+
 export function buildDepGraph(
   mavenGraph: MavenGraph,
   includeTestScope = false,
 ): DepGraph {
   const { rootId, nodes } = mavenGraph;
   const { pkgInfo: root } = getPkgInfo(rootId);
   const builder = new DepGraphBuilder({ name: 'maven' }, root);
-  const visited: string[] = [];
+  const visited: Record<string, boolean> = {};
```

Comment:
We don't care about the order of the visited list, so replaced with an obj because of faster times.
list.includes is O(n)
obj[myString] is O(1)